elastic/kibana
# [Discover] Extend toast API by fingerprintId, make use in search/state management to prevent toast storms

## Summary

This PR is introducing a `fingerprintId` in the toast API. It's an optional option. If it's set, it prevents adding of toasts of the same id, if they are already displayed. This can be considered as pacifier to prevent toast storms, which are multiple toasts of the same content displayed on the same page. The given PR already fixes 2 toast storms 

* If Discover is loaded with an invalid state in the URL, 2 identical toasts were displayed. 
* If Discover is loaded, and you get offline, there were 2 identical error messages. this get's even worse when an auto interval is configured:
![Bildschirmfoto 2022-11-23 um 08 55 48](https://user-images.githubusercontent.com/463851/203496051-a3ce436e-0178-4169-b895-b00a6693646a.png)

## Testing
* Load Discover
* Switch `network` to `offline` in Chrome's performance tab of the Developer tools
* Refresh the results
* There should be just 1 error displayed
* Configure a auto interval, like 5s
* No more toast storm, just a single message is displayed

### Checklist
